### PR TITLE
update SMTChecker tests with Z3 4.11.2

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

File: scripts/error_codes.py [hunk 1]
```diff
@@ -227,10 +227,10 @@ def examine_id_coverage(top_dir, source_id_to_file_names, new_ids_only=False):
             return False
 
     old_source_only_ids = {
-        "1584", "1823",
+        "1218", "1584", "1823",
         "1988", "2066", "2833", "3356",
         "3893", "3996", "4010", "4458", "4802",
-        "4902", "5272", "5622", "5798", "7128", "7400",
+        "4902", "5272", "5622", "5798", "5840", "7128", "7400",
```
Comment (leonardoalt) [inline scripts/error_codes.py:233]: These 2 error IDs were SMT errors when the solver couldn't prove something, which surprisingly doesn't happen anymore with the new z3! It's a nice development. We should anyway add some more complex tests that cause that in a future PR.

File: test/libsolidity/SMTCheckerTest.cpp [hunk 2]
```diff
@@ -38,10 +38,8 @@ SMTCheckerTest::SMTCheckerTest(string const& _filename): SyntaxTest(_filename, E
 		BOOST_THROW_EXCEPTION(runtime_error("Invalid SMT \"show unproved\" choice."));
 
 	m_modelCheckerSettings.solvers = smtutil::SMTSolverChoice::None();
-	auto const& choice = m_reader.stringSetting("SMTSolvers", "any");
-	if (choice == "any")
-		m_modelCheckerSettings.solvers = smtutil::SMTSolverChoice::All();
-	else if (choice == "none")
+	auto const& choice = m_reader.stringSetting("SMTSolvers", "z3");
+	if (choice == "none")
```
Comment (leonardoalt) [inline test/libsolidity/SMTCheckerTest.cpp:42]: This was changed to reflect the new behavior of "z3 is default now".

**Review by ekpyron — DISMISSED** [review]
Fine if tests pass.

**Review by ekpyron — DISMISSED** [review]
Still fine, if tests pass this time.

File: test/cmdlineTests/model_checker_invariants_all/err [hunk 3]
```diff
@@ -5,8 +5,8 @@ Warning: Return value of low-level calls not used.
   | 		^^^^^^^^^^^
 
 Info: Contract invariant(s) for model_checker_invariants_all/input.sol:test:
-((x <= 0) || true)
+(!(x >= 1) || true)
```
Comment (nikola-matic) [inline test/cmdlineTests/model_checker_invariants_all/err:8]: Just curious, why this change? `(x <= 0)` and `!(x >= 1)` are equal, right?

Comment (leonardoalt) [reply test/cmdlineTests/model_checker_invariants_all/err:8]: Yea, invariant and counterexample outputs are very nondeterministic, so the solver will choose an answer at its own discretion. As long as the results are correct you may get anything

**Review by nikola-matic — APPROVED** [review]

File: scripts/build_emscripten.sh [hunk 4]
```diff
@@ -34,7 +34,7 @@ else
     BUILD_DIR="$1"
 fi
 
-# solbuildpackpusher/solidity-buildpack-deps:emscripten-13
+# solbuildpackpusher/solidity-buildpack-deps:emscripten-14
```
Comment (cameel) [inline scripts/build_emscripten.sh:37]: There's a bug here. The hash below has not been updated. Looks like [the nightly build in solc-bin has been failing since Friday](https://github.com/ethereum/solc-bin/actions/runs/3560588983/jobs/5980711895) due to this:
```
-- Found Z3: /emsdk/upstream/emscripten/cache/sysroot/usr/lib/cmake/z3/Z3Config.cmake (found version "4.11.0.0") 
CMake Error at CMakeLists.txt:84 (message):
  SMTChecker tests require Z3 4.11.2 for all tests to pass.

  Build with -DSTRICT_Z3_VERSION=OFF if you want to use a different version.
  You can also use -DUSE_Z3=OFF to build without Z3.  In both cases use
  --no-smt when running tests.


-- Configuring incomplete, errors occurred!
```

Pinging @r0qs.

Comment (leonardoalt) [reply scripts/build_emscripten.sh:37]: True! Made a PR https://github.com/ethereum/solidity/pull/13757

Comment (r0qs) [reply scripts/build_emscripten.sh:37]: Thanks, forgot about it.
